typesafe/shared/src/main/scala/zio/config/typesafe/TypesafePropertyType.scala

```diff
@@ -0,0 +1,21 @@
+package zio.config.typesafe
+
+import com.typesafe.config._
+import zio.config._
+
+import scala.util.Try
+
+object TypesafePropertyType {
+  import PropertyType.PropertyReadError
+
+  object ConfigValuePropertyType extends PropertyType[String, ConfigValue] {
+    override def read(propertyValue: String): Either[PropertyReadError[String], ConfigValue] =
+      Try(ConfigFactory.parseString(propertyValue).resolve).map(_.root)
```

Comment:
I couldn't find an accessible `ConfigValue` parser so this is kind of a hack. First we try parsing the string as a `Config` and if it succeeds we extract the `ConfigObject` within the `Config`. If we fail we prepend a key so it essentially gets wrapped into an object and we try parsing it again. If it succeeds we extract the value with the key.

The reason we don't skip the first parse is that because a hocon object might not be wrapped within parenthesis and prepend a key to it would produce an invalid string for the second parse.